servers/quarkus-catalog/src/main/java/org/projectnessie/server/catalog/LakehouseConfigManagementImpl.java

```diff
@@ -0,0 +1,110 @@
+import org.projectnessie.catalog.files.config.ImmutableAdlsOptions;
+import org.projectnessie.catalog.files.config.ImmutableGcsOptions;
+import org.projectnessie.catalog.files.config.ImmutableS3Options;
+import org.projectnessie.catalog.service.api.LakehouseConfigManagement;
+import org.projectnessie.catalog.service.config.ImmutableCatalogConfig;
+import org.projectnessie.catalog.service.config.ImmutableLakehouseConfig;
+import org.projectnessie.catalog.service.config.LakehouseConfig;
+import org.projectnessie.catalog.service.objtypes.ImmutableLakehouseConfigObj;
+import org.projectnessie.catalog.service.objtypes.LakehouseConfigObj;
+import org.projectnessie.versioned.storage.common.exceptions.ObjNotFoundException;
+import org.projectnessie.versioned.storage.common.exceptions.ObjTooLargeException;
+import org.projectnessie.versioned.storage.common.persist.Persist;
+
+public class LakehouseConfigManagementImpl implements LakehouseConfigManagement {
+  private final Persist persist;
+  private final boolean dynamic;
+  private final LakehouseConfig staticLakehouseConfig;
+
+  public LakehouseConfigManagementImpl(
+      Persist persist, boolean dynamic, LakehouseConfig staticLakehouseConfig) {
+    this.persist = persist;
+    this.dynamic = dynamic;
+    this.staticLakehouseConfig = staticLakehouseConfig;
+  }
+
+  @Override
+  public LakehouseConfig currentConfig() {
+    if (!dynamic) {
+      return staticLakehouseConfig;
+    }
+
+    try {
+      // We could eventually save the persistence round-trip in a followup change.
+      LakehouseConfigObj obj = loadCurrent();
+      return obj.lakehouseConfig();
+    } catch (ObjNotFoundException e) {
+      return ImmutableLakehouseConfig.builder()
+          .catalog(ImmutableCatalogConfig.builder().build())
+          .s3(ImmutableS3Options.builder().build())
+          .gcs(ImmutableGcsOptions.builder().build())
+          .adls(ImmutableAdlsOptions.builder().build())
+          .build();
+    }
+  }
+
+  private LakehouseConfigObj loadCurrent() throws ObjNotFoundException {
+    return persist.fetchTypedObj(
+        LakehouseConfigObj.OBJ_ID, LakehouseConfigObj.OBJ_TYPE, LakehouseConfigObj.class);
+  }
+
+  @Override
+  public void updateConfig(LakehouseConfig config, LakehouseConfig expected) {
+    checkState(dynamic, "Cannot update the lakehouse config in Quarkus configuration");
+
+    checkArgument(
+        config != null && expected != null, "Updated config and expected must not be null");
+    checkArgument(!config.equals(expected), "Updated config and expected config must not be equal");
+
+    ImmutableLakehouseConfigObj newObj =
+        LakehouseConfigObj.builder()
+            .versionToken(UUID.randomUUID().toString())
+            .lakehouseConfig(config)
+            .build();
+
+    try {
+      while (true) {
+        try {
+          LakehouseConfigObj existing = loadCurrent();
+          checkArgument(
+              existing.lakehouseConfig().equals(expected),
```

Comment:
I'm not sure I completely follow the logic of this loop... `expected` is a loop invariant, so `existing.lakehouseConfig()` must be an invariant too. What does the loop achieve by `loadCurrent()`? Do we expect other processes to change the version of the `LakehouseConfigObj` without changing its `lakehouseConfig()`?

Reply:
Yea, it would likely fail in this assertion in case of two concurrent requests here.